pkgs/applications/misc/nnn/default.nix

```diff
@@ -1,48 +1,100 @@
-{ lib, stdenv, fetchFromGitHub, pkg-config, makeWrapper, ncurses, readline
-, archivemount, atool, fzf, libarchive, rclone, sshfs, unzip, vlock
-, conf ? null, withIcons ? false, withNerdIcons ? false }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, pkg-config
+, symlinkJoin
+, file
+, ncurses
+, readline
+, which
+# optional dependencies
+#   these can alternatively be picked up from PATH at runtime
+, withExtraTools ? false
+, archivemount
+, atool
+, fzf
+, libarchive
+, rclone
+, sshfs
+, unzip
+, vlock
+# options
+, conf ? null
+, withIcons ? false
+, withNerdIcons ? false
+}:
 
 # Mutually exclusive options
 assert withIcons -> withNerdIcons == false;
 assert withNerdIcons -> withIcons == false;
 
-stdenv.mkDerivation rec {
-  pname = "nnn";
-  version = "4.0";
+let
+  binPath = lib.makeBinPath ([
+    file
+    which
+  ]);
 
-  src = fetchFromGitHub {
-    owner = "jarun";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "0cbxgss9j0bvsp3czjx1kpm9id7c5xxmjfnvjyk3pfd69ygif2kl";
-  };
+  extraBinPath = lib.makeBinPath ([
+    archivemount
```

Comment:
archivemount is currently broken on Darwin, but would be fixed when #118269 is merged. But it won't matter much since this won't be required by default.

Reply:
No longer relevant.